addons/mail/models/discuss/discuss_poll_answer.py

```diff
@@ -0,0 +1,146 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.mail.tools.discuss import Store
+from odoo.exceptions import ValidationError
+from odoo.fields import Domain
+from odoo.tools import format_list
+
+
+class DiscussPollAnswer(models.Model):
+    _description = "Poll answer"
+    _name = "discuss.poll.answer"
+
+    number_of_votes = fields.Integer(compute="_compute_number_of_votes")
+    poll_id = fields.Many2one("discuss.poll", ondelete="cascade", required=True, index=True)
+    selected_by_self = fields.Boolean(compute="_compute_selected_by_self")
+    text = fields.Char(required=True)
+    vote_percentage = fields.Integer(compute="_compute_vote_percentage")
+    voter_guest_ids = fields.Many2many("mail.guest")
+    voter_user_ids = fields.Many2many("res.users")
```

Comment:
This could just be a 3-column table with an answer_id | partner_id | guest_id.
This will simplify all the SQL in this PR.